src/EventSubscriber/XLocationIdResponseSubscriber.php

```diff
@@ -0,0 +1,66 @@
+<?php
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace EzSystems\PlatformHttpCacheBundle\EventSubscriber;
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Event\FilterResponseEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+/**
+ * Rewrites the X-Location-Id HTTP header.
+ *
+ * This is a BC layer for custom controllers (including REST server) still
+ * using X-Location-Id header which is now deprecated. For
+ * full value of tagging, see docs/using_tags.md for how to take advantage of the
+ * system.
+ */
+class XLocationIdResponseSubscriber implements EventSubscriberInterface
+{
+    /**
+     * @var string
+     */
+    private $tagHeader;
+
+    public function __construct($tagHeader)
+    {
+        $this->tagHeader = $tagHeader;
+    }
+
+    public static function getSubscribedEvents()
+    {
+        return [KernelEvents::RESPONSE => ['rewriteCacheHeader', -5]];
+    }
+
+    public function rewriteCacheHeader(FilterResponseEvent $event)
+    {
+        $response = $event->getResponse();
+        if (!$response->headers->has('X-Location-Id')) {
```

Comment:
Nitpick: I'd make `'X-Location-Id'` a class constant.

Reply:
But it would be public 😢   Un indispensable?

Reply:
No, it was a simple nitpick. I don't think it's an issue that the constant is public, but if it itches your goat, you can skip it :)

Reply:
🐐

Reply:
On the other side, in this case it will never and can never change, so actually should be fine even for my 🐐  So will fix that on merge :)